### ISSUE TITLE
support resizing of parameters

### ISSUE THREAD

**Issue by redpony** [clab/cnn#32]
it should be possible to resize parameters after they are allocated.

**Comment by chunyang-wen** [comment]
Yeah. I think this is a useful function. Any update on this?

**Comment by neubig** [comment]
No, this is not trivial because of DyNet's memory management scheme.

One common request I get is that people want LookupParameters where we don't know the vocabulary size before hand. In this case a current work-around is to create a LookupParameter object with a large vocabulary that will definitely be bigger than your expected vocabulary size.

If we want to fix this issue, I see two options:
1. We could do something tricky within the current memory management paradigm, but I'm not sure what.
2. We could write an additional memory manager that supports deletion of memory blocks. This wouldn't be too hard, and would allow us to allocate a new memory block, copy the parameters, then delete the old memory block.
